poppy/tests/test_fresnel.py

```diff
@@ -293,56 +299,57 @@ def test_fresnel_optical_system_Hubble(display=False):
 
-
-    ### check the beam size is as expected at primary and secondary mirror
-    assert(np.allclose(waves[1].spot_radius().value, 1.2))
-    # can't find a definitive reference for the beam diam at the SM, but
-    # the secondary mirror's radius is 14.05 cm
-    # We find that the beam is indeed slightly smaller than that.
-    assert(waves[2].spot_radius() > 13*u.cm )
-    assert(waves[2].spot_radius() < 14*u.cm )
-
-    ### check the focal length of the overall system is as expected
-    expected_system_focal_length = 1./(1./fl_pri + 1./fl_sec - (d_pri_sec)/(fl_pri*fl_sec))
-    # n.b. the value calculated here, 57.48 m, is a bit less than the
-    # generally stated focal length of Hubble, 57.6 meters. Adjusting the
-    # primary-to-secondary spacing by about 100 microns can resolve this
-    # discrepancy. We here opt to stick with the values used in the PROPER
-    # example, to facilitate cross-checking the two codes.
-
-    assert(not np.isfinite(waves[0].focal_length))  # plane wave after circular aperture
-    assert(waves[1].focal_length==fl_pri)           # focal len after primary
-    # NB. using astropy.Quantities with np.allclose() doesn't work that well
-    # so pull out the values here:
-    assert(np.allclose(waves[2].focal_length.to(u.m).value,
-           expected_system_focal_length.to(u.m).value)) # focal len after secondary
-
-    ### check the FWHM of the PSF is as expected
-    measured_fwhm = utils.measure_fwhm(psf)
-    expected_fwhm = 1.028*0.5e-6/2.4*206265
-    # we only require this to have < 5% accuracy with respect to the theoretical value
-    # given discrete pixelization etc.
-    assert(np.abs((measured_fwhm-expected_fwhm)/expected_fwhm) < 0.05)
-
-    ### check the various plane types are as expected, including toggling into angular coordinates
-    assert_message = ("Expected FresnelWavefront at plane #{} to have {} == {}, but got {}")
-    system_planetypes = [PlaneType.pupil, PlaneType.pupil, PlaneType.intermediate, PlaneType.image]
-    for idx, (wavefront, planetype) in enumerate(zip(waves, system_planetypes)):
-        assert wavefront.planetype == planetype, assert_message.format(
-            idx, "planetype", plane_type, wavefront.planetype
-        )
-
-    angular_coordinates_flags = [False, False, False, True]
-    for idx, (wavefront, angular_coordinates) in enumerate(zip(waves, angular_coordinates_flags)):
-        assert wavefront.angular_coordinates == angular_coordinates, assert_message.format(
-            idx, "angular_coordinates", angular_coordinates, wavefront.angular_coordinates
-        )
-
-    spherical_flags = [False, True, True, False]
-    for idx, (wavefront, spherical) in enumerate(zip(waves, spherical_flags)):
-        assert wavefront.spherical == spherical, assert_message.format(
-            idx, "spherical", spherical, wavefront.spherical
-        )
+    if len(waves)>1:
```

Comment:
This `if` statement's a little cryptic. Do I infer correctly that you had to add it because the CUDA version otherwise would fail this test since it can't return intermediates?

Reply:
That's right, can added a comment to that effect.